client/app/cloud/project/compute/infrastructure/virtualMachine/addEdit/cloud-project-compute-infrastructure-virtualMachine-addEdit.controller.js

```diff
@@ -966,7 +966,11 @@ angular.module("managerApp")
 
     $scope.$watch('VmAddEditCtrl.model.sshKeyId', function (value, oldValue) {
         if  (value) {
-            self.vmInEdition.sshKey = _.find(self.panelsData.sshKeys, { id: value });
+            var sshkey = _.find(self.panelsData.sshKeys, { id: value });
+            if (!sshkey || !sshkey.availableOnRegion) {
+                return;
```

Comment:
not sure about return nothing.

Reply:
It's to make sure the rest of the logic is escaped.  It's hacky but everything in there is so no big deal.

Reply:
good. I approved.